compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -292,90 +291,120 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         &mut self,
         obligation: &PolyTraitObligation<'tcx>,
     ) -> Result<Vec<PredicateObligation<'tcx>>, SelectionError<'tcx>> {
-        use rustc_transmute::{Answer, Condition};
-        #[instrument(level = "debug", skip(tcx, obligation, predicate))]
+        use rustc_transmute::{Answer, Assume, Condition};
+
+        /// Generate sub-obligations for reference-to-reference transmutations.
```

Comment:
You should document or link to documentation to some sort of specification that explains when two references are safely transmutable.

Reply:
Or else, it would be nice to pull up these inline comments into some sort of bulleted list here.